pkg/gce-pd-csi-driver/cache.go

```diff
@@ -174,24 +191,161 @@ func setupCaching(devicePath string, req *csi.NodeStageVolumeRequest, nodeId str
 			"--force",
 			"-y",
 		}
-		info, err = common.RunCommand("" /* pipedCmd */, "" /* pipedCmdArg */, "lvconvert", args...)
+		info, err = common.RunCommand("" /* pipedCmd */, nil /* pipedCmdArg */, "lvconvert", args...)
 		if err != nil {
 			return mainDevicePath, fmt.Errorf("Errored while setting up caching for volume %s %w: %s", devicePath, err, info)
 		}
 	}
 
 	// activate all the LVs in the Volume group
-	info, err = common.RunCommand("" /* pipedCmd */, "" /* pipedCmdArg */, "vgchange", []string{"-ay", volumeGroupName}...)
+	info, err = common.RunCommand("" /* pipedCmd */, nil /* pipedCmdArg */, "vgchange", []string{"-ay", volumeGroupName}...)
 	if err != nil {
 		// The logical volumes would not be accessible if the group is not activated
 		return mainDevicePath, fmt.Errorf("Failed to activate volume group %v %v:%s", volumeGroupName, err, info)
 	}
 	return mainDevicePath, nil
 }
 
+func ValidateDataCacheConfig(dataCacheMode string, datacacheSize string, ctx context.Context, nodeName string) error {
+	if dataCacheMode != "" && datacacheSize != "" {
+		isAlreadyRaided, err := IsRaided()
+		if err != nil {
+			return fmt.Errorf("Local SSDs are not setup for caching; got error: %v", err)
+		}
+		if !isAlreadyRaided {
+			return fmt.Errorf("Local SSDs are not setup for caching")
+		}
+		return nil
+	}
+	klog.Infof("Data cache is not enabled for PVC")
+	return nil
+}
+
+func GetDataCacheCountFromNodeLabel(ctx context.Context, nodeName string) (int, error) {
+	if nodeName == common.TestNode {
+		return common.LocalSSDCountForDataCache, nil
+	}
+	cfg, err := rest.InClusterConfig()
+	// We want to capture API errors with node label fetching, so return -1
+	// in those cases instead of 0.
+	if err != nil {
+		return -1, err
+	}
+	kubeClient, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		return -1, err
+	}
+	node, err := kubeClient.CoreV1().Nodes().Get(ctx, nodeName, metav1.GetOptions{})
+	if err != nil {
+		// We could retry, but this error will also crashloop the driver which may be as good a way to retry as any.
+		return -1, err
+	}
+	if val, found := node.GetLabels()[fmt.Sprintf(common.NodeLabelPrefix, common.DataCacheLssdCountLabel)]; found {
+		dataCacheCount, err := strconv.Atoi(val)
+		if err != nil {
+			return -1, fmt.Errorf("Error getting Datacache's LSSD count from node label: %v", err)
+		}
+		klog.Infof("Number of local SSDs requested for Datacache: %v", dataCacheCount)
+		return dataCacheCount, nil
+	}
+	return 0, fmt.Errorf("Cannot get Datacache's LSSD count from node label")
```

Comment:
There are nodes do not have data cache disk configured, these nodes should not return an error?

Do we have unit tests coverage for these cases?